demo-server/templates/index-template.handlebars

```diff
@@ -76,6 +76,7 @@
         function initWidget(widget) {
           {{#if redirectless}}
           var config = {
+            // interactWith: PfAuthnWidget.INTERACT_WITH_USER_AUTHZ,
```

Comment:
I find the naming `interactWith` a little bit awkward and confusing. 

As an alternative, what about just allowing the developer to specify the (relative) authorization endpoint. We can provide constants for `/as/authorization.oauth2` and `/as/user_authz.oauth2`. And of course, the default if no value is provided would be `/as/authorization.oauth2`.

Reply:
To be a bit more clear, I'm envisioning we would replace the `interactWith` field with a `authzEndpoint` field, or similar.

Reply:
Using the authorization endpoint as an indicator of the type of authorization doesn't cover all the cases and we might face some issues. The first thing that comes to mind is path translation at the reverse proxy layer (e.g. /as/authorization.auth2 to /as/authorization and vice-versa). This can easily happen in PingFederate Since the User Authorization is customizable OOB. The authorization URL doesn't necessarily indicate what type of interaction the widget should do. 
With this in mind, I propose the following changes:

1. Rename "interactWith" to something simpler and meaningful.
2. Introduce an optional "authzEndpoint" for better flexibility (I'm surprised no one has asked for this yet)

Reply:
Okay, I see. Maybe an integer `flowType` then, instead of `interactWith` ?